### PR TITLE
Improve allcombinations docstring + minor cleanups after #3256

### PR DESCRIPTION
**Pull request by bkamins**
Make sure we are explicit which values vary fastest when constructing a data frame.

### CODE REVIEW

**Comment by bkamins** [comment]
I also cleaned up code after #3256.

File: src/abstractdataframe/abstractdataframe.jl [hunk 1]
```diff
@@ -985,11 +985,19 @@ julia> dropmissing(df, [:x, :y])
...
         # What column indices should disallowmissing be applied to
         cols_inds = BitSet(index(df)[cols])
-        
+
         use_threads = Threads.nthreads() > 1 && ncol(df) > 1 && length(selected_rows) >= 100_000
         @sync for (i, col) in enumerate(eachcol(df))
-            @spawn_or_run use_threads if disallowmissing && (i in cols_inds)
-                new_columns[i] = Missings.disallowmissing(Base.view(col, selected_rows))
+            @spawn_or_run use_threads if disallowmissing && (i in cols_inds) &&
+                          (Missing <: eltype(col) && !(Any <: eltype(col)))
```
Comment (nalimilan) [inline src/abstractdataframe/abstractdataframe.jl:992]:
```suggestion
                          (Missing <: eltype(col) && eltype(col) !== Any)
```

File: src/abstractdataframe/abstractdataframe.jl [hunk 2]
```diff
@@ -985,11 +985,19 @@ julia> dropmissing(df, [:x, :y])
...
+                # Perform this path only if column eltype allows missing values
+                # except Any, as nonmissingtype(Any) == Any.
+                # Under these conditions Missings.disallowmissing must allocate
+                # a fresh column
+                col_sel = Base.view(col, selected_rows)
+                new_col = Missings.disallowmissing(col_sel)
+                @assert newcol !== col_sel
```
Comment (nalimilan) [inline src/abstractdataframe/abstractdataframe.jl:999]:
```suggestion
                @assert new_col !== col_sel
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**